From 36f0e7c4ec07dc6941ed9842a3e0664cf341ccc7 Mon Sep 17 00:00:00 2001
From: Laur59 <148864407+Laur59@users.noreply.github.com>
Date: Sat, 27 Sep 2025 20:17:21 +0300
Subject: [PATCH 3/3] Adapt CMake configuration to needs for uKOS-X

Use the runtimes project to build compiler-rt

(cherry picked from commit c77c0ded3475577be92660e28335499201d48f79)
---
 arm-software/embedded/CMakeLists.txt              | 5 +++--
 arm-software/embedded/arm-runtimes/CMakeLists.txt | 9 +++++++--
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/arm-software/embedded/CMakeLists.txt b/arm-software/embedded/CMakeLists.txt
index fe9879857888..755a8e9b5be9 100644
--- a/arm-software/embedded/CMakeLists.txt
+++ b/arm-software/embedded/CMakeLists.txt
@@ -151,7 +151,7 @@ set(PARALLEL_LIB_BUILD_LEVELS
 option(
     ENABLE_QEMU_TESTING
     "Enable tests that use QEMU. This option is ON by default."
-    ON
+    OFF
 )
 option(
     ENABLE_FVP_TESTING
@@ -228,6 +228,7 @@ set(LLVM_DISTRIBUTION_COMPONENTS
     llvm-strings
     llvm-strip
     llvm-symbolizer
+    scan-build
     LTO
     CACHE STRING ""
 )
@@ -242,7 +243,7 @@ installed by the install-llvm-toolchain target"
 )
 set(LLVM_ENABLE_PROJECTS clang;lld CACHE STRING "")
 set(LLVM_TARGETS_TO_BUILD AArch64;ARM CACHE STRING "")
-set(LLVM_DEFAULT_TARGET_TRIPLE aarch64-unknown-linux-gnu CACHE STRING "")
+set(LLVM_DEFAULT_TARGET_TRIPLE arm-none-eabi CACHE STRING "")
 set(LLVM_APPEND_VC_REV OFF CACHE BOOL "")
 set(LLVM_ENABLE_TERMINFO OFF CACHE BOOL "")
 set(LLVM_ENABLE_ZSTD OFF CACHE BOOL "")
diff --git a/arm-software/embedded/arm-runtimes/CMakeLists.txt b/arm-software/embedded/arm-runtimes/CMakeLists.txt
index eff0caf6ef94..66a68f621264 100644
--- a/arm-software/embedded/arm-runtimes/CMakeLists.txt
+++ b/arm-software/embedded/arm-runtimes/CMakeLists.txt
@@ -274,7 +274,7 @@ if(ENABLE_COMPILER_RT_TESTS)
         -DCOMPILER_RT_EMULATOR=${lit_test_executor}
         -DCOMPILER_RT_TEST_COMPILER=${LLVM_BINARY_DIR}/bin/clang
         -DCOMPILER_RT_TEST_COMPILER_CFLAGS=${compiler_rt_test_flags}
-        -DLLVM_INCLUDE_TESTS=ON
+        -DLLVM_INCLUDE_TESTS=OFF
         -DLLVM_LIT_ARGS=${compiler_rt_lit_args}
         -DLLVM_EXTERNAL_LIT=${external_lit_path}
     )
@@ -414,6 +414,7 @@ if(C_LIBRARY STREQUAL picolibc)
 
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/meson-cross-build.txt.in ${CMAKE_CURRENT_BINARY_DIR}/meson-cross-build.txt @ONLY)
 
+    set(PICOLIBC_OPTIONS -Dmultilib=false -Dio-c99-formats=true -Dio-long-long=true -Dpicolib=true -Dnewlib-multithread=true -Dposix-io=true -Dpicocrt=false -Dpicocrt-lib=false -Dtests=false)
     ExternalProject_Add(
         picolibc
         STAMP_DIR ${PROJECT_PREFIX}/picolibc/${VARIANT_BUILD_ID}/stamp
@@ -429,6 +430,7 @@ if(C_LIBRARY STREQUAL picolibc)
             -Dincludedir=include
             -Dlibdir=lib
             -Dspecsdir=none
+            ${PICOLIBC_OPTIONS}
             -Dmultilib=false
             -Ddebug=false
             -Dtests-enable-stack-protector=false
@@ -553,7 +555,10 @@ if(C_LIBRARY MATCHES "^newlib")
             --disable-nls
             --enable-lite-exit
             --disable-multilib
-            --enable-newlib-retargetable-locking)
+            --enable-newlib-retargetable-locking
+            --enable-newlib-io-float)
+        set(newlib_optim_flags
+            "-Os")
     endif()
 
     include(${TOOLCHAIN_SOURCE_DIR}/cmake/fetch_newlib.cmake)
-- 
2.50.1 (Apple Git-155)

